frontend/src/components/lists/value-filters.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Menu from '../common/menu';
+import { fetchRequests, fetchTimesOfDay } from '../../redux/actions';
+import { STATUS_LOADING } from '../../constants';
+
+export function FilterFieldValues({ allPossibleValues, values, onChange, onOpen, loading }) {
+    function getUpdatedValues(value) {
+        if(values.includes(value)) {
+            return values.filter(v => v !== value);
+        }
+
+        return [...values, value];
+    }
+
+    const label = values.length > 0 ? `(${values.length})` : '';
+
+    return <div className="filter-field">
+        <Menu
+            alignLeft={true}
+            icon="filter"
+            className="icon"
+            loading={loading}
+            onOpen={onOpen}
+            options={allPossibleValues.map(value =>
+                <React.Fragment>
+                    <input
+                        type="checkbox"
+                        onChange={() => onChange(getUpdatedValues(value))}
+                        checked={values.includes(value)}
+                    ></input>
+                    <label>{value}</label>
+                </React.Fragment>    
+            )}
+        />
+        {label}
+    </div>;
+}
+
+export function FilterTimeOfDay() {
+    const dispatch = useDispatch();
```

Comment:
I've use the [Redux hooks API](https://react-redux.js.org/api/hooks) just to avoid prop tunnelling. I don't think we use it anywhere else so don't mind doing that for consistency though

Reply:
Nope - more than happy to keep this code clean!